builtin/credential/aws/path_role.go

```diff
@@ -20,6 +20,12 @@ func pathRole(b *backend) *framework.Path {
 				Type:        framework.TypeString,
 				Description: "Name of the role.",
 			},
+			"allowed_auth_types": {
```

Comment:
There is also a question of which path to follow when both types are supported and all the required parameters are set. I am inclined towards making this a value that takes in just a single value. Then a role will allow just one flow and it will be easier to reason about what happens when. That way, this can default to "ec2" to retain old behavior and have operators explicitly set "iam" when needed. Unfortunately, in that case, we can't default to "iam".